src/platform/plugins/shared/data/common/search/expressions/esql.ts

```diff
@@ -181,8 +181,12 @@ export const getEsqlFn = ({ getStartDependencies }: EsqlFnArguments) => {
         })
       ).pipe(
         switchMap(({ search, uiSettings }) => {
+          // this is for backward compatibility, if the query is of fields or functions type
+          // and the query is not set with ?? in the query, we should set it
+          // https://github.com/elastic/elasticsearch/pull/122459
+          const fixedQuery = fixESQLQueryWithVariables(query, input?.esqlVariables ?? []);
```

Comment:
Should there be some kind of e2e test or something to make sure this never gets removed? Maybe that's overkill 🤷‍♂️

Reply:
I think it is an overkill Drew, variables are on tech preview so even if someone removed it (which means that we miss it during the review) the panel will fail but these things are expected for tech preview features.

Reply:
If this isn’t a forever thing, what you say totally makes sense.